tst/com/amazon/corretto/crypto/provider/test/EvpSignatureSpecificTest.java

```diff
@@ -98,6 +138,48 @@ private static void testKeyTypeMismatch(final String algorithm, final String bas
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("ecdsaBadSignatures")
+    @SuppressWarnings("unchecked")
+    public void testBadEcdsaSignature(final String algorithm, final BigInteger r, final BigInteger s)
+            throws GeneralSecurityException {
+        final Signature verifier = Signature.getInstance(algorithm, NATIVE_PROVIDER);
+        // We always use the largest key to avoid hash truncation confusion
+        final ECPublicKey pubKey = (ECPublicKey) ECDSA_PAIR.getPublic();
+        final byte[] signature;
+        byte[] rArr = r.toByteArray();
+        if (rArr[0] == 0) { // Trim leading zero byte if present
```

Comment:
why does this (sometimes?) have a leading zero? is that an ASN.1 thing?

Reply:
`BigInteger` returns signed values and these values are all positive, so must have a signed zero bit. Thus if the order of the curve fits exactly into a multiple of 8 bits, the leading one bit of the order with be the first bit of a byte, and thus when we add a sign bit of 0, it overflows to a new byte.

Reply:
makes sense, thanks.  last piece of the puzzle for me was this [javadoc](https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html#toByteArray()) stating that `toByteArray` always returns big-endian, 2's complement byte array.